### DISCUSSION TITLE
Testing / developing on LazyVim

### DISCUSSION THREAD

**Discussion by askeko** [LazyVim/LazyVim#1092] [Q&A] [answered]
This might be a bit of an odd / newbie question, but it's been puzzling me for a bit while trying to make it work on my own:

How do you develop on LazyVim in terms of testing before you push to upstream? 
I'm asking because I wanted to start from scratch in terms of creating my own nvim setup, and i like the split of a publicly available standard that can then be configured locally, like LazyVim, nvChad, lunarvim, etc. I've used some minimum code from LazyVim to have it working, but in order to test each change I either have to push to upstream, or, in my own config, source each mention of the LazyVim plugin (e.g. when setting a new colorscheme option, and in the lazy.lua file) as well as lazyvim.plugins.core file in the LazyVim plugin, to my local clone location using "dir = ..".

The latter works ok, but it is slightly inconvenient to change the names at these locations every time I want to test/implement features. Also I'm _lazy_ and would like to know how to do it more conveniently.

Cheers

**Comment by Mayrixon** [comment]
Here is my solution.

```lua
-- ~/debug_lazyvim.lua

-- bootstrap lazy
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)

require("lazy").setup({
	spec = {
		-- add LazyVim and import its plugins
		{ dir = "~/workspace/Lazyvim/", import = "lazyvim.plugins" }, -- modify LazyVim fork path here
		-- import any extras modules here
		-- { import = "lazyvim.plugins.extras.lang.typescript" },
		-- { import = "lazyvim.plugins.extras.lang.json" },
		-- { import = "lazyvim.plugins.extras.ui.mini-animate" },
		-- import/override with your plugins
		-- { import = "plugins" },
	},
	install = { colorscheme = { "tokyonight", "habamax" } },
	checker = { enabled = true }, -- automatically check for plugin updates
	performance = {
		rtp = {
			-- disable some rtp plugins
			disabled_plugins = {
				"gzip",
				-- "matchit",
				-- "matchparen",
				-- "netrwPlugin",
				"tarPlugin",
				"tohtml",
				"tutor",
				"zipPlugin",
			},
		},
	},
})

vim.cmd.colorscheme("tokyonight")
```

This script is based on [LazyVim starter](https://github.com/LazyVim/starter) and uses the same data and state files as the `normal` nvim.
What you need to do is:

1. Modify the path to your fork of LazyVim (on Line 23).
2. Comment out the configurations about LazyVim in the file `lazyvim/plugins/core.lua`.
3. Start Neovim with the command `nvim -u ~/debug_lazyvim.lua`.

**Reply by askeko** [reply]
That's pretty much what I've been doing I think, however it seem a little clunky having to make these changes every time you need to implement and test if a new feature works. Seems like it could be done in a simpler way.. hopefully. Will see what jyuan0 replies or eventually test the #1129 answer when I'm back at my dev system.

**Comment by jyuan0** [comment, marked as answer]
I'm not sure I understand your question, but I think this might help: https://github.com/LazyVim/LazyVim/discussions/1129

**Reply by askeko** [reply]
Not able to test right now, but the way it is described in the answer seems like what i want to achieve. To clarify, the answer provides a way to change the LazyVim code locally and test the impact of the changes (nvim uses the local repo rather than upstream)?

**Reply by dpetka2001** [reply]
Yes, exactly that.

**Reply by askeko** [reply]
Thanks! Guess I should be better at searching the repo for answers next time, dunno why I missed that.

**Reply by jyuan0** [reply]
Glad it helped you.

**Reply by dpetka2001** [reply]
You probably missed that, cuz now that I look at the dates my issue was created later than yours :stuck_out_tongue:
